tests/func/test_data_status.py

```diff
@@ -146,7 +123,10 @@ def test_skip_uncached_pipeline_outputs(tmp_dir, dvc, run_copy_metrics):
 
 def test_output_with_newly_added_stage(tmp_dir, dvc):
     dvc.stage.add(deps=["bar"], outs=["foo"], name="copy", cmd="cp foo bar")
-    assert dvc.data_status() == {**EMPTY_STATUS, "not_in_cache": ["foo"]}
+    # assert dvc.data_status() == {**EMPTY_STATUS, "not_in_cache": ["foo"]}
+    # TODO: discuss what the output should be
+    assert dvc.data_status() == EMPTY_STATUS
+    assert dvc.data_status(granular=True) == EMPTY_STATUS
```

Comment:
Minor, but something that we should clarify. Let's take following `dvc.yaml` file as an example:
```yaml
# dvc.yaml
stages:
  stage1:
    cmd: cp foo bar
    deps: [bar]
    outs: [foo]
```

Let's say, that you added this to `dvc.yaml` and immediately run this. What would you expect to see? Currently, it skips them. Do we want to show them as `deleted` and/or maybe `Not in cache` too? Or, skipping is fine? 
cc @dberenbaum

---

Let's generalize now. This issue occurs if we have the entry for outputs in the `dvc.yaml` file or any `.dvc` file, but we don't have any hashes (i.e. no `dvc.lock` for the stage, or no `md5/etag/checksum` specified in `.dvc` file).

Reply:
I think skipping is fine for now. In general, the most typical use case is a new `dvc.yaml` file, followed by options like `add --no-exec` or `import-url --no-download`. In those cases, it's expected that the file is not in the cache and is missing, so it might be more confusing to show those statuses. There are cases where it could be useful, like someone who forgets to commit `dvc.lock`, but I think those are less common than the new `dvc.yaml` file/stage.